test/unit/authority_view.cpp

```diff
@@ -30,7 +30,7 @@ class authority_view_test
         {
             authority_view a;
             BOOST_TEST(a.empty());
-            BOOST_TEST(a.size() == 0);
+            BOOST_TEST_EQ(a.size(), std::size_t(0));
```

Comment:
why not just `0` ?

Reply:
`0` emits lots of warnings because they are different types. We could implement something similar to [`std::cmp_equal`](https://en.cppreference.com/w/cpp/utility/intcmp) for the test operators, but the best solution is to directly use the proper types with `0u`, as all other boost libraries do.